wallet-new/server/Cardano/Wallet/API/V1/Handlers/Wallets.hs

```diff
@@ -57,20 +59,27 @@ listWallets PaginationParams {..} = do
       }
     _ -> return $ OneOf $ Left example
 
-updatePassword :: WalletId
-               -> PasswordUpdate
-               -> MonadV1 Wallet
+updatePassword
+    :: WalletId
+    -> PasswordUpdate
+    -> MonadV1 Wallet
 updatePassword _ _ = liftIO $ generate arbitrary
 
-deleteWallet :: WalletId
-             -> MonadV1 NoContent
-deleteWallet _ = return NoContent
+-- | Deletes an exisiting wallet.
+deleteWallet
+    :: (HasConfigurations, MonadWalletLogic ctx m)
```

Comment:
As you are calling `deleteWallet` from the main `handlers`, you don't need the extra `HasConfigurations`. Have you tried removing that?

Reply:
Correct, I should have removed that once I switched from `MonadV1` because `MonadWalletLogic` contains it:
```
type MonadWalletLogic ctx m =
    ( MonadWalletLogicRead ctx m
    , MonadWalletDB ctx m
    , MonadKeys m
    )
```
```
type MonadWalletLogicRead ctx m =
    ( MonadIO m
    , MonadThrow m
    , WithLogger m
    , MonadRandom m
    , MonadSlots ctx m
    , MonadDBRead m
    , MonadBalances m
    , MonadWalletDBRead ctx m
    , MonadKeysRead m
    , MonadTxpMem WalletMempoolExt ctx m  -- TODO: remove these two once 'fixingCachedAccModifier' becomes useless
    , BlockLockMode ctx m
    , HasConfiguration
    )
```